crates/nu-protocol/src/eval_base.rs

```diff
@@ -292,6 +291,7 @@ pub trait Eval {
                 Ok(Value::glob(pattern, *quoted, expr.span))
             }
             Expr::MatchBlock(_) // match blocks are handled by `match`
+            | Expr::Block(_) // blocks are handled directly by core commands
```

Comment:
When you pull stuff out of `Call` in e.g. `if`/`for` etc. you are not affected by this right.

Reply:
Raw `Expression`s are pulled out of `Call` without any eval, so yeah this shouldn't affect it.